src/Workspaces/Remote/ServiceHub/Host/RemoteAnalyzerAssemblyLoaderService.cs

```diff
@@ -3,39 +3,28 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
-using System.Composition;
+using System.Collections.Generic;
 using System.Collections.Immutable;
-using System.Diagnostics;
+using System.Composition;
 using System.IO;
-using System.Reflection;
 using Microsoft.CodeAnalysis.Host;
 using Microsoft.CodeAnalysis.Host.Mef;
-using System.Collections.Generic;
 
-namespace Microsoft.CodeAnalysis.Remote.Diagnostics
-{
-    /// <summary>
-    /// Customizes the path where to store shadow-copies of analyzer assemblies.
-    /// </summary>
-    [ExportWorkspaceService(typeof(IAnalyzerAssemblyLoaderProvider), [WorkspaceKind.RemoteWorkspace]), Shared]
-    internal sealed class RemoteAnalyzerAssemblyLoaderService : IAnalyzerAssemblyLoaderProvider
-    {
-        private readonly RemoteAnalyzerAssemblyLoader _loader;
-        private readonly ShadowCopyAnalyzerAssemblyLoader _shadowCopyLoader;
-
-        [ImportingConstructor]
-        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-        public RemoteAnalyzerAssemblyLoaderService([ImportMany] IEnumerable<IAnalyzerAssemblyResolver> externalResolvers)
-        {
-            var baseDirectory = Path.GetDirectoryName(Path.GetFullPath(typeof(RemoteAnalyzerAssemblyLoader).GetTypeInfo().Assembly.Location));
-            Debug.Assert(baseDirectory != null);
+namespace Microsoft.CodeAnalysis.Remote.Diagnostics;
 
-            var resolvers = externalResolvers.ToImmutableArray();
-            _loader = new(baseDirectory, resolvers);
-            _shadowCopyLoader = new(Path.Combine(Path.GetTempPath(), "VS", "AnalyzerAssemblyLoader"), resolvers);
-        }
+/// <summary>
+/// Customizes the path where to store shadow-copies of analyzer assemblies.
+/// </summary>
+[ExportWorkspaceService(typeof(IAnalyzerAssemblyLoaderProvider), [WorkspaceKind.RemoteWorkspace]), Shared]
+[method: ImportingConstructor]
+[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+internal sealed class RemoteAnalyzerAssemblyLoaderService(
+    [ImportMany] IEnumerable<IAnalyzerAssemblyResolver> externalResolvers)
+    : IAnalyzerAssemblyLoaderProvider
+{
+    private readonly ShadowCopyAnalyzerAssemblyLoader _shadowCopyLoader =
+        new(Path.Combine(Path.GetTempPath(), "VS", "AnalyzerAssemblyLoader"), externalResolvers.ToImmutableArray());
```

Comment:
Do you use this even on Linux? On Linux we don't even need to shadow copy, we can just load from stream. 

Mostly just curious about this question.

Reply:
on linux we have no OOP.  it's strictly a windows thing due to VS being .net fx, and wanting a real .net core process to run in :)